modules/lvgl/lvgl.c

```diff
@@ -219,6 +219,34 @@ static int lvgl_allocate_rendering_buffers(lv_display_t *display)
 }
 #endif /* CONFIG_LV_Z_BUFFER_ALLOC_STATIC */
 
+#ifdef CONFIG_LV_Z_HANDLER_THREAD
+
+K_SEM_DEFINE(lv_timer_sem, 0, 1);
+
+static void lvgl_timer_handler_expired(struct k_timer *timer)
+{
+	ARG_UNUSED(timer);
+	k_sem_give(&lv_timer_sem);
+}
+
+K_TIMER_DEFINE(lv_handler_timer, lvgl_timer_handler_expired, NULL);
+
+static void lvgl_timer_handler_thread_entry(void *arg1, void *arg2, void *arg3)
+{
+	while (1) {
+		k_sem_take(&lv_timer_sem, K_FOREVER);
+		uint32_t reload_value = lv_timer_handler();
+		/*schedule next timer verification*/
+		k_timer_start(&lv_handler_timer, K_MSEC(MIN(reload_value, INT32_MAX)), K_MSEC(0));
```

Comment:
I suppose the semaphore we still need to ensure that `lvgl_init` is called beforehand, but why would we need the timer? I think having a simple sleep here is essentially the same - allowing another thread to run, also I think it does not give us anything precision wise as well.

Reply:
I was asking the same thing until I compared the sleep loop with timer based approach, the main advantage here is the timing slippage is better compensated on virtual timers, this reflects for example on having rock solid 30ms without perceptible
jitter when running the demo benchmark.

The soft timer usage approach will show up more interesting when running at 60Hz of the refresh rate.

So here we get an extra ROM+RAM overhead in exchange of having a better frame counter :)